### ISSUE TITLE
SD-Card full error message with over 2 GB free storage space

### ISSUE THREAD

**Issue by Codeberg-AsGithubAlternative-buhtz** [ankidroid/Anki-Android#6441]
This issue is based on [this discussion on the anki-droid mailinglist](https://groups.google.com/d/msgid/anki-android/062aa4cc-23f6-16d2-55a1-8acad2edc9f5%40gmail.com).

Problems started with verson 2.11.0 but are still reproducable with current stable release 2.11.1.

Device: Old Samsung Galaxy (S2 or S3) "GT-I8190" with Android 4.1.2.
My Anki runed for years on that device without any problems like this.

I always get the message

> "SD-Karte voll
> Sicherung nicht gespeichert.
> Nicht genug Speicherplatz auf der SD-Karste. Geben Sie mehr
> Speicherplatz frei oder verringern Sie die Sicherungstiefe." 

The message appears when starting Anki.

When I first got that message (with Anki 2.11.0) I had round about 0,4 GB free space on my device - which is usual for me. Anki had no problems with that in the past. My deck (on AnkiWeb) is round about 16 MB.
I cleaned up some files so now I have round about 2,3 GB free space on the device.
There is no SD-Card in the device - just using the in-build-memory.

In the advanced settings the "AnkiDroid-Directory"
(AnkiDroid-Verzeichnis) points to `/sdcard/AnkiDroid`. This path do not
exist.
So I checked my file system and found a `/storage/sdcard0/AnkiDroid`
folder. I think `sdcard0` is the in-build-storage. I point Anki to that
folder. But the error is still there.

I reduced the backup deepness ("Sicherungstiefe") to 1 and to 0 also. Error is still there even if the value is 0.

I deinstalled Anki and fully deleted the `AnkiDroid` folder. Restarted, reinstalled. Problem is still there. The folder is recreated with nearly empty files - as expected. Here is a `collection.log` for example
```
[1591950560] Collection.java:<init>():
/storage/sdcard0/AnkiDroid/collection.anki2,  2.11.1
[1591950560] Sched.java:_updateCutoff(): 0,  1592013600
```

I connected AnkiDroid to my ankiweb, fully synced. Error appears again (after sync).
I "checked database" - no problems.
Error message is still there.

Beside this error message I still can use Anki as normal. Entering my deck, learn the cards, syncing.

But I am scarred about loseing my data.

**Comment by mikehardy** [comment]
This is great information, thank you! For reference, you can always grab the older releases (2.10.x, I think 2.10.3 was the latest 2.10.x release) so that you can review

Then if you want to help continue troubleshooting this you can grab a "parallel release" for 2.11.2 (about to release), like 2.11.2.A from the releases page, that way you can run them side by side - 2.10.x to get work done, 2.11.x-A to see what's wrong with 2.11.x

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
I am not sure what do you want. ;)

Do I have to test if the error appears in the older 2.10 versions?

And do I have to test if the error still appears in the current alpha?

**Comment by mikehardy** [comment]
You don't *have* to do anything. I was suggesting a way to:

- unblock your ability to review cards (i.e., use the app!) with https://github.com/ankidroid/Anki-Android/releases/download/v2.10.4/AnkiDroid-2.10.4.apk
- still help troubleshoot, in case you were interested with https://github.com/ankidroid/Anki-Android/releases/download/v2.11.2/AnkiDroid-2.11.2.parallel.A.apk

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
Checking `2.10-A`...
Error message appears.

Checking `2.11.2-B` ...
Error message appears.

btw: I really like your solution with the "parallel" naming.

**Comment by mikehardy** [comment]
That's unexpected, you say 2.10-A the error appears but you also say that the problem started with 2.11 :thinking: 
Could always try a parallel version of 2.9.6? https://github.com/ankidroid/Anki-Android/releases/tag/v2.9.6

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
I think this indicates that the reason for the error is on my side. So it is not a modification fro 2.10 to 2.11.

I did something. I do not know what.

Maybe I remind me wrong. It is possible that I did some clean up on my device that causing this problem. I can not rule it out. You often use a smartphone in situations where you are not highly concentrated. ;)

Can we find the right code location to see what can cause such messages?

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
![shot_2020-06-15_13-34-27](https://user-images.githubusercontent.com/11861496/84653214-4ed15a00-af0d-11ea-9031-f314b7fde315.png)

**Comment by david-allison-1** [comment]
To be fixed in #6468

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
Can confirm this is fixed in 2.11.2 from today. Thanks a lot!
Have not checked the current alpha because it seams fixed in the current stable also.
